src/odemis/driver/xt_client.py

```diff
@@ -164,6 +297,63 @@ def __init__(self, name, role, children, address, daemon=None,
                 self._detector = Detector(parent=self, daemon=daemon, **ckwargs)
             self.children.value.add(self._detector)
 
+    def transfer_latest_package(self, data: bytes) -> None:
+        """
+        Transfer the latest xtadapter package.
+
+        Note:
+            Pyro has a 1 gigabyte message size limitation.
+            https://pyro5.readthedocs.io/en/latest/tipstricks.html#binary-data-transfer-file-transfer
+
+        :param data: The package's zip file data in bytes.
+
+        """
+        with self._proxy_access:
+            self.server._pyroClaimOwnership()
+            return self.server.transfer_latest_package(data)
+
+    def check_and_transfer_latest_package(self) -> None:
+        """Check if a latest xtadapter package is available and then transfer it."""
+        try:
+            package = None
+            bitness = re.search("bitness:\s*([\da-z]+)", self._swVersion)
+            bitness = bitness.group(1) if bitness is not None else None
+            adapter = "xtadapter"
+            if "xttoolkit" in self._swVersion:
+                adapter = "fastem-xtadapter"
+            current_version = re.search("xtadapter:\s*([\d.]+)", self._swVersion)
+            current_version = current_version.group(1) if current_version is not None else None
+            if current_version is not None and bitness is not None:
+                package = check_latest_package(
+                    directory=XT_INSTALL_DIR,
+                    current_version=current_version,
+                    adapter=adapter,
+                    bitness=bitness,
+                    is_zip=True,
+                )
+            if package is not None:
+                # Check if it's a proper zip file
+                zip_file = zipfile.ZipFile(package.path)
+                ret = zip_file.testzip()
+                zip_file.close()
+                if ret is None:
+                    # Open the package's zip file as bytes and transfer them
+                    with open(package.path, mode="rb") as f:
+                        data = f.read()
+                    self.transfer_latest_package(data)
+                    notify2.init(__name__)
```

Comment:
The backend uses "Odemis" as the name in the notifications. It might be better to use the same, so that notifications are grouped together. (or at least, there is a chance that this happens).